backend/inspirehep/orcid/utils.py

```diff
@@ -226,3 +233,37 @@ def distributed_lock(lock_name, expire=10, auto_renewal=True, blocking=False):
 
 class DistributedLockError(Exception):
     pass
+
+
+def update_moved_orcid(old_orcid, new_orcid):
+    try:
+        author_record = AuthorsRecord.get_record_by_pid_value(old_orcid, "orcid")
+        if new_orcid not in get_value(author_record, "ids.value"):
+            new_author_ids = [
+                {"schema": "ORCID", "value": new_orcid},
+                *author_record["ids"],
+            ]
+            author_record["ids"] = new_author_ids
+            author_record.update(dict(author_record))
+    except PIDDoesNotExistError:
+        LOGGER.info(
+            "ORCID account was moved, but no profile matching ORCID number found on INSPIRE",
+            orcid=old_orcid,
+        )
+    user_orcid_account = RemoteAccount.query.filter(
+        RemoteAccount.user_id == UserIdentity.id_user, UserIdentity.id == old_orcid
+    ).one_or_none()
+    if user_orcid_account:
+        remove_access_token_for_orcid_account(user_orcid_account, old_orcid)
+    LOGGER.info("ORCID update", new_orcid=new_orcid, old_orcid=old_orcid)
+    db.session.commit()
+
+
+def remove_access_token_for_orcid_account(user_orcid_account, orcid):
+    access_token = RemoteToken.query.filter(
+        RemoteToken.id_remote_account == user_orcid_account.id
+    ).first()
```

Comment:
a relation has been set up, so you can simply do
```suggestion
    access_token = user_orcid_account.remote_tokens[0]
```